flathunter/crawl_immobilienscout.py

```diff
@@ -88,28 +96,38 @@ def get_entries_from_javascript(self):
 
     def get_entries_from_json(self, json):
         """Get entries from JSON"""
-        jsonpath_expr = parse("$..['resultlist.realEstate']")
         return [
-          self.extract_entry_from_javascript(entry.value) for entry in jsonpath_expr.find(json)
+            self.extract_entry_from_javascript(entry.value) for entry in self.JSON_PATH_PARSER_ENTRIES.find(json)
         ]
 
     def extract_entry_from_javascript(self, entry):
         """Get single entry from JavaScript"""
-        image_path = parse("$..galleryAttachments..['@xlink.href']")
+
+        # the url that is being returned to the frontend has a placeholder for screen size. (%WIDTH% and %HEIGHT%)
+        # The website's frontend fills these variables based on the user's screen size.
+        # If we remove this part, the API will return the original size of the image.
+        #
+        # Before:
+        # https://pictures.immobilienscout24.de/listings/$$IMAGE_ID$$.jpg/ORIG/legacy_thumbnail/%WIDTH%x%HEIGHT%3E/format/webp/quality/50
+        #
+        # After: https://pictures.immobilienscout24.de/listings/$$IMAGE_ID$$.jpg
+
+        images = [
+            '/'.join(image.value.split('/')[:5]) for image in self.JSON_PATH_PARSER_IMAGES.find(entry)
+        ]
+
```

Comment:
```suggestion
        entry_id = int(entry.get("@id", 0))

        return {
            'id': entry_id,
            'url': "https://www.immobilienscout24.de/expose/" + entry_id,
```

Reply:
int and string cannot be combined. but thanks for the suggestion.